xgboost_ray/tests/test_colocation.py

```diff
@@ -0,0 +1,95 @@
+import os
+import shutil
+import tempfile
+from unittest.mock import patch
+import numpy as np
+
+import ray
+from xgboost_ray import train, RayDMatrix, RayParams
+from xgboost_ray.main import _train
+from xgboost_ray.tests.test_fault_tolerance import _kill_callback
```

Comment:
nit - we should avoid making tests a module -- it can cause weird dependency chains.

Reply:
I moved these callbacks to the util file.